packages/store/src/utils/store-validators.ts

```diff
@@ -14,4 +17,32 @@ export abstract class StoreValidators {
       throw new Error(this.stateNameErrorMessage(name));
     }
   }
+
+  public static getValidStateNamesMap(states: StateClass[]): Map<StateName, StateClassName> {
+    const uniqueStateNamesMap: Map<StateName, StateClassName> = new Map();
+
+    for (let i = 0, size: number = states.length; i < size; i++) {
```

Comment:
maybe call it `length` - more semantic maybe? and also do you need to specify the `number` type here?